feat: update to forge-build-tools 3.2.0 with async minifyHtml API

tyler-josh opened the pull request:
## Summary
- Updated forge-build-tools to version 3.2.0
- Implements async minifyHtml API for improved performance
- Maintains API compatibility and build stability

## Changes
- feat: update to forge-build-tools 3.2.0 with async minifyHtml API

## Testing
- Build verification completed
- No breaking changes introduced

tyler-josh committed:
feat: update to forge-build-tools 3.2.0 with async minifyHtml API

BREAKING CHANGE: Updates forge-build-tools dependency to 3.2.0 which includes
async minifyHtml() API. This change is transparent to CLI users but may affect
any code that directly imports internal utilities.

Security fixes included:
- Addresses html-minifier ReDoS vulnerability (GHSA-pfq8-rq6v-vf5m)
- Addresses lodash.template command injection (GHSA-35jh-r3h4-6jhm)
- Addresses @octokit ReDoS vulnerabilities via npm overrides
- Updates brace-expansion to address ReDoS vulnerability

CLI interface remains fully compatible.

tyler-josh committed:
revert: remove version bump and changelog - let CI handle versioning

DRiFTy17 commented on .gitignore:
```diff
@@ -3,3 +3,4 @@ dist/
 test/coverage/
 .vscode/
 .DS_Store
+CLAUDE.md
```
Could you undo the changes to this file? We're going to add claude context files at some point across all of the repos.

DRiFTy17 commented on CHANGELOG.md:
```diff
@@ -1,3 +1,4 @@
+
```
I don't think it'll mess with the changelog generator, but might as well just remove the unrelated line change here

DRiFTy17 commented on bin/forge.js:
What changed with this file?

tyler-josh committed:
fix: revert gitignore changes to allow team CLAUDE.md guidance

- Remove CLAUDE.md from .gitignore to allow team to add official guidance
- Local CLAUDE.md remains untracked for development use
